cmd/dagger/module.go

```diff
@@ -1093,6 +1108,30 @@ func (o *modObject) IsCore() bool {
 	return o.SourceModuleName == ""
 }
 
+func skipFunction(obj, field string) bool {
+	// TODO: make this configurable in the API but may not be easy to
+	// generalize because an "internal" field may still need to exist in
+	// codegen, for example. Could expose if internal via the TypeDefs though.
+	skip := map[string][]string{
+		"Query": {
+			// for SDKs only
+			"builtinContainer",
+			"generatedCode",
+			"currentFunctionCall",
+			"currentModule",
+			"typeDef",
+			// for tests only
+			"secret",
+			// deprecated
+			"pipeline",
+		},
+	}
```

Comment:
These are the hardcoded exclusions when you do `dagger core --help`. May want to add more to the list. To be clear, this means these commands won't be created. It's not just for listing in `--help`.

Example of the output with current list in this PR's description.

Reply:
I think these should be excluded as well:
- set-secret
- cache-volume

I can see a use case for the remaining ones even if just to debug, introspect, or experiment/prototype.